browser/src/__tests__/NodeAutosuggest.test.tsx

```diff
@@ -1,22 +1,49 @@
-import {shallow, configure} from 'enzyme';
+import { configure} from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16'
 import {expect} from 'chai';
 import NodeAutoSuggestService from "../service/NodeAutosuggestService"
-import React from 'react';
 
 
 
 configure({adapter: new Adapter()});
 
 describe("Map of class to Provider has correct short name", () => {
-    test("should display a blank login form, with remember me checked by default", async () => {
-      // ???
-    });
 
-    test('sum 1 + 2 = 3' , () => {
-      const nodeContainer = shallow(<NodeAutoSuggestService/>);
-      expect((nodeContainer.instance() as NodeAutoSuggestService).add(1,2)).equals(3)
+    test('test 1', () =>{
+      const nodeAutoSuggest = new NodeAutoSuggestService();
+      expect(nodeAutoSuggest.parseString(
+        "java.util.Map<java.lang.Class<? extends com.snap.durablejob.DurableJob<?>>,javax.inject.Provider<"+
+        "com.snap.durablejob.DurableJobProcessor<?,?>>>")).equals(
+        "Map<Class<DurableJob<?>> Provider<DurableJobProcessor<?>>>"
+        )
     })
 
+    test('test 2', () => {
+      const nodeAutoSuggest = new NodeAutoSuggestService();
+      expect(nodeAutoSuggest.parseString("com.snap.dagger.browser.sample.CoffeeApp.CoffeeShop")).equals(
+        "CoffeeApp.CoffeeShop")
+    })
+
+    test('test 3', () => {
+      const nodeAutoSuggest = new NodeAutoSuggestService();
+      expect(nodeAutoSuggest.parseString(
+        "java.util.Map<java.lang.Class<? extends com.snap.durablejob.DurableJob<?>>,javax.inject.Provider<"+
+        "com.snap.durablejob.DurableJobProcessor<?,?>>> com.snap.lenses.app.LensesDurableJobComponent_MultibindingsModule"+
+        "#provide_socialUnlockResponseCacheCleanupProcessorMultibinding")).equals(
+           "Map<Class<DurableJob<?>> Provider<DurableJobProcessor<?>>> LensesDurableJobComponentMultibindingsModule"
+          )
+    })
+
+    test('test 4', () => {
+      const nodeAutoSuggest = new NodeAutoSuggestService();
+      expect(nodeAutoSuggest.parseString("@io.plaidapp.core.dagger.DesignerNewsApi okhttp3.OkHttpClient")).equals(
+        "DesignerNewsApiOkHttpClient")
```

Comment:
```@DesignerNewsApiOkHttpClient OkHttpClient```

Reply:
here, we find the qualifier,  convert its type to a simple display name,  and then add it to the beginning of the result as `@[simpleName]`